### PR TITLE
Sneha Bagri | Issue-5 | Read Shopping list suggestion JSON data and s…

### PR DESCRIPTION
**Pull request by bsneha90**
…how it to the user

Issue Link - https://github.com/akhil-ghatiki/foreCastAlexSkillBackend/compare/SuggestedList?expand=1

### CODE REVIEW

File: src/main/java/com/thoughtworks/forecastAlexaSkillBackend/domain/ShoppingListReader.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+
+public class ShoppingListReader {
+    public ShoppingList suggestedList() {
+        String jsonPath= "../suggestedShoppingList.json";
+        BufferedReader bufferedReader = null;
+        try {
+            bufferedReader = new BufferedReader(new FileReader(jsonPath));
+        } catch (FileNotFoundException e) {
```
Comment (akhil-ghatiki) [inline src/main/java/com/thoughtworks/forecastAlexaSkillBackend/domain/ShoppingListReader.java:21]: Can we log the exception here ?

File: src/main/java/com/thoughtworks/forecastAlexaSkillBackend/domain/ShoppingListReader.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        return parse(bufferedReader);
+    }
+
+    private ShoppingList parse(BufferedReader metadataReader) {
+        try {
+            ShoppingList shoppingList = new Gson().fromJson(metadataReader, ShoppingList.class);
+            return shoppingList;
+        } catch (Exception e) {
+            return null;
+        }
```
Comment (akhil-ghatiki) [inline src/main/java/com/thoughtworks/forecastAlexaSkillBackend/domain/ShoppingListReader.java:35]: We need to close the buffered reader.

File: src/main/java/com/thoughtworks/forecastAlexaSkillBackend/handlers/LaunchRequestHandler.java [hunk 3]
```diff
@@ -20,7 +22,7 @@ public boolean canHandle(HandlerInput input) {
 
     //process the file, get count and kgs variable and into into this string
     //String speechText = "Hello User, {count} people have purchased {kgs} of sugar. Do u want to place order as well ?";
-    String speechText = "Hello Minnions!!";
+    String speechText = "Hello Minnions!! DO You want to go over your grocery list? ";
```
Comment (akhil-ghatiki) [inline src/main/java/com/thoughtworks/forecastAlexaSkillBackend/handlers/LaunchRequestHandler.java]: "DO". Just a small typo.

File: src/main/java/com/thoughtworks/forecastAlexaSkillBackend/model/ShoppingItem.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
+package com.thoughtworks.forecastAlexaSkillBackend.model;
+
+import com.google.gson.internal.$Gson$Types;
+
+public class ShoppingItem {
```
Comment (akhil-ghatiki) [inline src/main/java/com/thoughtworks/forecastAlexaSkillBackend/model/ShoppingItem.java:8]:
```suggestion
@Getter
@Setter
public class ShoppingItem {
```

We can use the getter and setter annotations here. that will avoid all the methods that we need to write below for getting and setting

**Review by akhil-ghatiki — CHANGES_REQUESTED** [review]
Can we commit the Json file as well ?

File: src/main/java/com/thoughtworks/forecastAlexaSkillBackend/domain/ShoppingListReader.java (hunk 2 above)
Comment (bsneha90) [reply src/main/java/com/thoughtworks/forecastAlexaSkillBackend/domain/ShoppingListReader.java:35]: Done

File: src/main/java/com/thoughtworks/forecastAlexaSkillBackend/handlers/LaunchRequestHandler.java (hunk 3 above)
Comment (bsneha90) [reply src/main/java/com/thoughtworks/forecastAlexaSkillBackend/handlers/LaunchRequestHandler.java]: Done

File: src/main/java/com/thoughtworks/forecastAlexaSkillBackend/model/ShoppingItem.java (hunk 4 above)
Comment (bsneha90) [reply src/main/java/com/thoughtworks/forecastAlexaSkillBackend/model/ShoppingItem.java:8]: done

File: src/main/java/com/thoughtworks/forecastAlexaSkillBackend/domain/ShoppingListReader.java (hunk 1 above)
Comment (bsneha90) [reply src/main/java/com/thoughtworks/forecastAlexaSkillBackend/domain/ShoppingListReader.java:21]: done

**Review by SwathiVarkala — APPROVED** [review]

**Review by akhil-ghatiki — APPROVED** [review]

**Pull request merged**